java/maven.junit.ui/src/org/netbeans/modules/maven/junit/ui/MavenJUnitNodeOpener.java

```diff
@@ -144,6 +159,24 @@ public void run(CompilationController compilationController) throws Exception {
             UIJavaUtils.openFile(fo2open[0], (int) line[0]);
         }
     }
+    
+    private Element findInnermostTypeElement(Element containingTypeElement, List<? extends String> targetInnerClassNames) {
```

Comment:
Yes. I'm not sure how to test this, since this class is glued to some static methods. I'll take a look, but if you could point to a test of similar code elsewhere, it would be helpful.

Reply:
@JaroslavTulach are there tests of similar code that could be used as a starting point for testing this code?

Reply:
No need to stub `CompilationController:` refactor code to have the tested part callable and save make accessors so the test can collect the computed data. I.e. make the UserTask a package-private member class, so the test can instantiate it. Create sources in test data, then process it using JavaSource.parse(). Call the tested code with (real) CompilationController instance and observe the outcome.